orchagent/neighorch.cpp

```diff
@@ -1227,6 +1229,434 @@ bool NeighOrch::removeNeighbor(const NeighborEntry &neighborEntry, bool disable)
     return true;
 }
 
+/**
+ * @brief Creates a neighbor add entry and adds it to bulker.
+ * @param ctx NeighborBulkContext contains neighbor information and list of object statuses.
+ */
+bool NeighOrch::addBulkNeighbor(NeighborBulkContext& ctx)
+{
+    SWSS_LOG_ENTER();
+
+    sai_status_t status;
+    auto& object_statuses = ctx.object_statuses;
+
+    const MacAddress &macAddress = ctx.mac;
+    const NeighborEntry neighborEntry = ctx.neighborEntry;
+    string alias = neighborEntry.alias;
+    IpAddress ip_address = neighborEntry.ip_address;
+
+    SWSS_LOG_INFO("Adding neighbor entry %s on %s to bulker.", ip_address.to_string().c_str(), alias.c_str());
+
+    sai_object_id_t rif_id = m_intfsOrch->getRouterIntfsId(alias);
+    if (rif_id == SAI_NULL_OBJECT_ID)
+    {
+        SWSS_LOG_INFO("Failed to get rif_id for %s", alias.c_str());
+        return false;
+    }
+
+    sai_neighbor_entry_t neighbor_entry;
+    neighbor_entry.rif_id = rif_id;
+    neighbor_entry.switch_id = gSwitchId;
+    copy(neighbor_entry.ip_address, ip_address);
+
+    vector<sai_attribute_t> neighbor_attrs;
+    sai_attribute_t neighbor_attr;
+
+    neighbor_attr.id = SAI_NEIGHBOR_ENTRY_ATTR_DST_MAC_ADDRESS;
+    memcpy(neighbor_attr.value.mac, macAddress.getMac(), 6);
+    neighbor_attrs.push_back(neighbor_attr);
+
+    if ((ip_address.getAddrScope() == IpAddress::LINK_SCOPE) && (ip_address.isV4()))
+    {
+        /* Check if this prefix is a configured ip, if not allow */
+        IpPrefix ipll_prefix(ip_address.getV4Addr(), 16);
+        if (!m_intfsOrch->isPrefixSubnet (ipll_prefix, alias))
+        {
+            neighbor_attr.id = SAI_NEIGHBOR_ENTRY_ATTR_NO_HOST_ROUTE;
+            neighbor_attr.value.booldata = 1;
+            neighbor_attrs.push_back(neighbor_attr);
+        }
+    }
+
+    PortsOrch* ports_orch = gDirectory.get<PortsOrch*>();
+    auto vlan_ports = ports_orch->getAllVlans();
+
+    for (auto vlan_port: vlan_ports)
```

Comment:
This is a lot of repeated code from existing `addNeighbor`. I think we should only add the bulk function as discussed offline.

Reply:
Reduced the amount of repeated code in latest commit